velox/experimental/cudf/exec/CudfHashAggregation.cpp

```diff
@@ -485,8 +527,12 @@ auto toAggregators(
     auto const inputIndex = aggInputs[0];
     auto const constant = aggConstants.empty() ? nullptr : aggConstants[0];
     auto const companionStep = getCompanionStep(kind, step);
-    aggregators.push_back(
-        createAggregator(companionStep, kind, inputIndex, constant, isGlobal));
+    const auto originalName = getOriginalName(kind);
+    const auto resultType = exec::isPartialOutput(companionStep)
+        ? exec::Aggregate::intermediateType(originalName, argumentTypes)
+        : exec::Aggregate::finalType(originalName, argumentTypes);
```

Comment:
Since companion function is also registered as a separate function, is it necessary to get the original name? Shouldn't the result of `exec::Aggregate::intermediateType(kind)` give the correct output type? Are you intentionally overriding the companion function's output type with the original function's type based on plan node step?